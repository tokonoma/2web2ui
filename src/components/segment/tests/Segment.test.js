import React from 'react';
import { shallow } from 'enzyme';
import Segment from '../Segment';
import getConfig from 'src/helpers/getConfig';

jest.mock('src/helpers/getConfig');

describe('Segment', () => {
  const subject = () => shallow(<Segment />);

  it('does not add the script when segment config is enabled: false', () => {
    getConfig.mockReturnValue(false);
    const wrapper = subject();

    expect(wrapper).toMatchSnapshot();
  });

  it('adds the script when segment config is enabled: true', () => {
    getConfig.mockReturnValueOnce(true).mockReturnValueOnce('abcdefg1234567');
    const wrapper = subject();

    expect(wrapper).toMatchSnapshot();
  });
});
