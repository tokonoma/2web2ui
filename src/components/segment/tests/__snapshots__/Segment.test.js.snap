// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment adds the script when segment config is enabled: true 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    type="text/javascript"
  >
    !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e&lt;analytics.methods.length;e++){var t=analytics.methods[e];analytics[t]=analytics.factory(t)}analytics.load=function(e,t){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+e+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=t};analytics.SNIPPET_VERSION="4.1.0";
analytics.load("abcdefg1234567");
analytics.page();
}}();
  </script>
</HelmetWrapper>
`;

exports[`Segment does not add the script when segment config is enabled: false 1`] = `""`;
